synapse/app/generic_worker.py

```diff
@@ -738,6 +738,10 @@ async def _process_and_notify(self, stream_name, instance_name, token, rows):
         except Exception:
             logger.exception("Error processing replication")
 
+    async def on_position(self, stream_name: str, instance_name: str, token: int):
+        await super().on_position(stream_name, instance_name, token)
+        await self.on_rdata(stream_name, instance_name, token, [])
```

Comment:
This probably needs a comment, but I'm not really sure what to suggest...

> Also call `on_rdata` to ensure that stream positions are properly reset.

Or something?

Reply:
Yeah, something like that. I don't have any better suggestions at least :/